src/TrieSet.mo

```diff
@@ -26,14 +26,18 @@ module {
 
   /// Put an element into the set.
   public func put<T>(s : Set<T>, x : T, xh : Hash, eq : (T, T) -> Bool) : Set<T> {
-    let (s2, _) = Trie.put<T,()>(s, { key = x; hash = xh }, eq, ());
-    s2
+    // let (s2, _) = Trie.put<T,()>(s, { key = x; hash = xh }, eq, ());
+    // s2
+    s := Trie.put<T,()>(s, { key = x; hash = xh x}, eq, ()).0;
+    s
   };
 
   /// Delete an element from the set.
   public func delete<T>(s : Set<T>, x : T, xh : Hash, eq : (T, T) -> Bool) : Set<T> {
-    let (s2, _) = Trie.remove<T, ()>(s, { key = x; hash = xh }, eq);
-    s2
+    // let (s2, _) = Trie.remove<T, ()>(s, { key = x; hash = xh }, eq);
+    // s2
+    s := Trie.remove<T, ()>(s,  { key = x; hash = xh}, eq).0;
```

Comment:
Dies this even compile? Function parameters are supposed to be immutable.

Reply:
Fixed.